lindeck-frontend/src/views/boody/deckPage/DeckPage.jsx

```diff
@@ -0,0 +1,128 @@
+import "./DeckPage.css"
+import 'react-grid-layout/css/styles.css';
+import React, {Component} from 'react';
+import ReactGridLayout from "react-grid-layout";
+import ReactResizeDetector from 'react-resize-detector';
+import MovingDeckEditButton from "./deckEditMenu/DeckEditMenu";
+import DeckEditDialog from "./editDeckDialog/DeckEditDialog";
+import Card from "./card/Card";
+import GS from "../../../common/classes/GlobalStorage";
+import {Redirect} from "@reach/router";
+
+
+class deckPage extends Component {
+    state = {
+        deck: GS.getDeckFromUsernameDeckname(this.props.username, this.props.deckname),
+        editDeckDialogOpened: false,
+        gridWidth: 800,
+    }
+
+    addCard = (card) => {
+        let newDeck = this.state.deck
+        newDeck.addCard(card)
+        GS.saveDeck(newDeck)
+        this.setState({deck: newDeck})
+    }
+
+    resetLayout = () => {
+        let newDeck = this.state.deck
+        newDeck.resetLayout()
+        GS.saveDeck(newDeck)
+        this.setState({deck: newDeck})
+    }
+
+    clearAll = () => {
+        let newDeck = this.state.deck
+        newDeck.clear()
+        GS.saveDeck(newDeck)
+        this.setState({deck: newDeck})
+    }
+
+    duplicateCard = (card) => {
+        this.state.deck.duplicateCard(card)  // will  call onLayoutChange, dntknw wy but it is
+    }
+
+    deleteCard = (id) => {
+        this.state.deck.deleteCard(id) // mb because of moving deck edit menu (if it is open, layoutChange does not calling)
+    }
+
+    onLayoutChange(layout) {
+        let newDeck = this.state.deck
+        newDeck.layout = layout
+        GS.saveDeck(newDeck)
+        this.setState({deck: newDeck});
+    }
+
+    changeCardAnswer = (text, id) => {
+        this.state.deck.getCard(id).answer = text
+        GS.saveDeck(this.state.deck)
+    }
+
+    changeCardVerdict = (verdict, id) => {
+        this.state.deck.getCard(id).verdict = verdict
+        GS.saveDeck(this.state.deck)
+        this.setState({deck: this.state.deck});
+    }
+
+    changeCardText = (text, id, isFlipped) => {
```

Comment:
It's bad, bad it can't be fixed  right now.

Reply:
waht? that your deck is not divided to Metadata + Deck?